dev/WindowsAppRuntime_MSIXInstallFromPath/main.cpp

```diff
@@ -161,29 +169,37 @@ HRESULT JustDoIt(PCWSTR path) noexcept try
         auto packageFullName{ std::wstring(line.c_str() + offset + 1) };
         THROW_HR_IF_MSG(E_UNEXPECTED, packageFullName.empty(), "line:%s", lineUtf8.c_str());
 
-        AddPackageIfNecessary(path, filename, packageFullName);
+        AddPackageIfNecessary(path, filename, packageFullName, forceDeployment);
+    }
+
+    // Restart Push Notifications Long Running Platform when ForceDeployment option is applied.
+    if (forceDeployment)
+    {
+        wil::com_ptr_nothrow<INotificationsLongRunningPlatform> longRunningProcessPlatform{
+            wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
```

Comment:
Does PushNotifications have an API? Would be preferable to assuming "restart push == CCI(pushserver)"

SUGGESTION:

1. `dev\PushNotificationsServicing.h` declares namespace Microsoft::blahblah::PushNotifications { inline HRESULT StartService() { return CCI(pushserver); }`
2. `devPushNotifications.vcxitems publishes `PushNotificationsServicing.h` (much like DynDep does for msixdynamicdependency.h)
3. Installer `#include <PushNotificationsServicing.h>` and calls `StartService()`

This way if/when Push' implementation evolves Push has direct control and visibility to the implementation and installer can be oblivious to the fine print

Reply:
Check my PR update for this which I coded before seeing your comment. If there is strong resistance to the method I used, I will add another commit. Looking to code complete.

Reply:
I wouldn't say strong or resistance. Merely a suggestion for refactoring - sooner or later. Not blocking here, just a good idea for long term hygiene